### ISSUE TITLE
Using the TypeScript Type System in TypeScript Projects

### ISSUE THREAD

**Issue by AdamSobieski** [microsoft/TypeScript#45093]
Hello. I am working on a project which needs a type system. It occurred to me, as the TypeScript type system is solid and includes the needed functionality for the project (e.g., type intersections and unions), to try to import and utilize the TypeScript type system.

```ts
import { Type } from 'typescript';

export interface TypedObject
{
    get type(): Type;
}
```

So far so good. Unfortunately, it is difficult to find online discussions about this scenario. Unfortunately, this scenario is not documented. I would like to request documentation about importing and using the TypeScript type system in TypeScript projects. Thank you.

**Comment by orta** [comment]
You can learn about how to use the typescript compiler API in the wiki: https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API

**Comment by AdamSobieski** [comment]
@orta, thank you. I reviewed the Compiler API documentation and browsed the source code.

Using `factory`, it appears that one can create objects which implement `TypeNode`-related interfaces.

The `TypeChecker` interface provides a function `getTypeFromTypeNode()` which appears to return `Type`s for input `TypeNode`s.

To get a `TypeChecker` object, it appears to require a `Program` as this interface provides a function `getTypeChecker()`.

I found the global functions `createProgram()` which provide objects implementing the `Program` interface.

I wonder whether it is possible to create instances of `Type` in a manner resembling the following. I wonder how, if it is possible, one would best go about implementing these three functions:

```
function createType(/* parameters */) : Type
{
  // ?
}
function createIntersectionType(types: readonly Type[]) : IntersectionType
{
  // ?
}
function createUnionType(types: readonly Type[]) : UnionType
{
  // ?
}
```

What I am hoping to do is to disentangle the TypeScript type system from compiling programs and to utilize it. Thank you.
